src/data/automation_i18n.ts

```diff
@@ -3,10 +3,281 @@ import { computeStateName } from "../common/entity/compute_state_name";
+      ? hass.states[trigger.at] || trigger.at
+      : trigger.at;
+
+    return `When the time is equal to ${at}`;
+  }
+
+  // Time Patter Trigger
+  if (trigger.platform === "time_pattern") {
+    return "Time Pattern Trigger";
+  }
+
+  // Zone Trigger
+  if (trigger.platform === "zone" && trigger.entity_id && trigger.zone) {
+    let entities = "";
+    let zones = "";
+    let zonesPlural = false;
+
+    const states = hass.states;
+
+    if (Array.isArray(trigger.entity_id)) {
+      for (const [index, entity] of trigger.entity_id.entries()) {
+        if (states[entity]) {
+          entities += `${index > 0 ? "," : ""} ${
+            trigger.entity_id.length > 1 &&
+            index === trigger.entity_id.length - 1
+              ? "or"
+              : ""
+          } ${computeStateName(states[entity]) || entity}`;
+        }
+      }
+    } else {
+      entities = states[trigger.entity_id]
+        ? computeStateName(states[trigger.entity_id])
+        : trigger.entity_id;
+    }
+
+    if (Array.isArray(trigger.zone)) {
+      if (trigger.zone.length > 1) {
+        zonesPlural = true;
+      }
+
+      for (const [index, zone] of trigger.zone.entries()) {
+        if (states[zone]) {
+          zones += `${index > 0 ? "," : ""} ${
+            trigger.zone.length > 1 && index === trigger.zone.length - 1
+              ? "or"
+              : ""
+          } ${computeStateName(states[zone]) || zone}`;
+        }
+      }
+    } else {
+      zones = states[trigger.zone]
+        ? computeStateName(states[trigger.zone])
+        : trigger.zone;
+    }
+
+    return `When ${entities} ${trigger.event}s ${zones} ${
+      zonesPlural ? "zones" : "zone"
+    }`;
+  }
+
+  // Geo Location Trigger
+  if (trigger.platform === "geo_location" && trigger.source && trigger.zone) {
+    let sources = "";
+    let zones = "";
+    let zonesPlural = false;
+    const states = hass.states;
+
+    if (Array.isArray(trigger.source)) {
+      for (const [index, source] of trigger.source.entries()) {
+        sources += `${index > 0 ? "," : ""} ${
+          trigger.source.length > 1 && index === trigger.source.length - 1
+            ? "or"
+            : ""
+        } ${source}`;
+      }
+    } else {
+      sources = trigger.source;
+    }
+
+    if (Array.isArray(trigger.zone)) {
+      if (trigger.zone.length > 1) {
+        zonesPlural = true;
+      }
+
+      for (const [index, zone] of trigger.zone.entries()) {
+        if (states[zone]) {
+          zones += `${index > 0 ? "," : ""} ${
+            trigger.zone.length > 1 && index === trigger.zone.length - 1
+              ? "or"
+              : ""
+          } ${computeStateName(states[zone]) || zone}`;
+        }
+      }
+    } else {
+      zones = states[trigger.zone]
+        ? computeStateName(states[trigger.zone])
+        : trigger.zone;
+    }
+
+    return `When ${sources} ${trigger.event}s ${zones} ${
+      zonesPlural ? "zones" : "zone"
+    }`;
+  }
+
```

Comment:
Can we add a little placeholder for Webhook, MQTT & Template? Looks a bit ugly now:

```suggestion
  // MQTT Trigger
  if (trigger.platform === "mqtt") {
    return "When a MQTT payload has been received";
  }
  
  // Template Trigger
  if (trigger.platform === "template") {
    return "When a template triggers";
  }
  
  // Template Trigger
  if (trigger.platform === "webhook") {
    return "When a Webhook payload has been received";
  }
```